README.md

```diff
@@ -12,6 +12,65 @@ _A collection of Android libraries to build browsers or browser-like application
 
 We encourage you to participate in this open source project. We love Pull Requests, Bug Reports, ideas, (security) code reviews or any kind of positive contribution. Please read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
 
+# Architecture and Overview
```

Comment:
Maybe we should move this to docs/architecture.md?

I also wonder if we should add a README.md (as a kind of table of contents) to docs/ and then link to that from the root README.md.

What do you think?

Reply:
Yes, thought about that too, but it's also nice to have everything right there when users navigate to the Repo. Of course, there will come a point where it's too much. I am not sure if we're already at that point :).